### PR TITLE
Add missing code examples on `LocalKey`

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @Amanieu

### CODE REVIEW

File: library/std/src/thread/local.rs [hunk 1]
```diff
@@ -452,7 +478,7 @@ impl<T: 'static> LocalKey<RefCell<T>> {
     /// Panics if the key currently has its destructor running,
     /// and it **may** panic if the destructor has previously been run for this thread.
     ///
-    /// # Example
+    /// # Examples
```
Comment (joboet) [inline library/std/src/thread/local.rs:481]: But it's still only one example?

**Review by joboet — APPROVED** [review]
The new examples make sense to me. r=me if you want to undo the "example" → "examples" change

File: library/std/src/thread/local.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply library/std/src/thread/local.rs:481]: It's how it's done in all std/core docs. I can't find where the rule is described though... Take a look at the [String docs](https://doc.rust-lang.org/nightly/std/string/struct.String.html) for example.

So even if there is only one code example, it's supposed to be titled "Examples".

**Comment by joboet** [comment]
Mmmh, yeah, fair enough.

@bors r+ rollup

**Pull request merged**